include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+						dbesc($photo_id),
+						dbesc($album));
+			} else {
+				$nothingtodo = true;
+			}
+
+			if(x($_FILES,'media')) {
+				$nothingtodo = false;
+				$media = $_FILES['media'];
+				$data = SaveMediaToDatabase("photo", $media, $type, $album, $allow_cid, $deny_cid, $allow_gid, $deny_gid, $desc, 0, $visibility, $photo_id);
+				if (!is_null($data))
+					return api_format_data("photo_update", $type, $data);
+			}
+
+			// return success of updating or error message
+			if ($result) {
+				$answer = array('result' => 'updated', 'message' => 'Image id `' . $photo_id . '` has been updated.');
+				return api_format_data("photo_update", $type, array('$result' => $answer));
+			}
+			else {
+				if ($nothingtodo) {
+					$answer = array('result' => 'cancelled', 'message' => 'Nothing to update for image id `' . $photo_id . '`.');
+					return api_format_data("photo_update", $type, array('$result' => $answer));
+				}
+				throw new BadRequestException("unknown error");
+			}
+		}
+		throw new BadRequestException("unknown error");
+	}
+
+
+	/**
+	 * @brief delete a single photo from the database through api
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photo_delete($type) {
+		if (api_user()===false)
+			throw new ForbiddenException();
+
+		// input params
+		$photo_id = (x($_REQUEST, 'photo_id') ? $_REQUEST['photo_id'] : null);
+
+		// do several checks on input parameters
+		// we do not allow calls without photo id
+		if ($photo_id == null)
+			throw new BadRequestException("no photo_id specified");
+
+		// check if photo is existing in database
+		$r = q("SELECT * FROM `photo` WHERE `uid` = %d AND `resource-id` = '%s'",
+				intval(api_user()),
+				dbesc($photo_id)
+			);
+		if (!dbm::is_result($r))
+			throw new BadRequestException("photo not available");
+
+		// now we can perform on the deletion of the photo
+		$result = q("DELETE FROM `photo` WHERE `uid` = %d AND `resource-id` = '%s'",
+				intval(api_user()),
+				dbesc($photo_id));
+
+		// return success of deletion or error message
+		if ($result) {
+			// retrieve the id of the parent element (the photo element)
+			$photo_item = q("SELECT `id` FROM `item` WHERE `uid` = %d AND `resource-id` = '%s' AND `type` = 'photo'",
+				intval(local_user()),
+				dbesc($photo_id)
+				);
+
+			if (!dbm::is_result($photo_item))
+				throw new InternalServerErrorException("problem with deleting items occured");
+
+			// function for setting the items to "deleted = 1" which ensures that comments, likes etc. are not shown anymore
+			// to the user and the contacts of the users (drop_items() do all the necessary magic to avoid orphans in database and federate deletion)
+			drop_item($photo_item[0]['id'],false);
+
+			$answer = array('result' => 'deleted', 'message' => 'photo with id `' . $photo_id . '` has been deleted from server.');
+			return api_format_data("photo_delete", $type, array('$result' => $answer));
 		}
+		else {
```

Comment:
Standards: Please move the `else` on the same line as the preceding closing brace.